controllers/authControllerRegisterController.test.js

```diff
@@ -0,0 +1,450 @@
+import { registerController } from './authController';
+import userModel from '../models/userModel.js';
+
+// Mock functions
+jest.mock('./../helpers/authHelper', () => ({
+    hashPassword: jest.fn(async (password) => Promise.resolve(`Mock ${password}`)),
+}));
+
+jest.mock('../models/userModel', () => {
+    //Need to mock the constructor because the constructor is used in registerController
+    const mockSave = jest.fn().mockResolvedValue({
+        name: "James",
+        email: "james",
+        password: "james123",
+        phone: "9123 4567",
+        address: "Sentosa",
+        answer: "Badminton"
+    });
+
+    return jest.fn().mockImplementation(() => ({
+        save: mockSave,
+    }));
+})
+
+
+// Test suite for registerController
+describe('registerController', () => {
+    let res;
+    
+    beforeEach(() => {
+        jest.clearAllMocks();
+
+        //Mock the status and send functions
+        res = {
+            send: jest.fn(),
+            status: jest.fn().mockReturnThis()
+        };
+    });
+
+   
+    describe('should return error messages for its input validations if', () => {
+        //NEVER PASS
+        it('name is empty', async () => {
```

Comment:
For this test and the ones below, not sure if It would be better to fill up the values that you are not testing (in this case fill up all the values except name)

Reply:
Let me think about this. For now, I think can leave it because because on the control flow of the function, when `name` is empty, it does not really matter what values the other variables are, since control flow will stop when `name` is empty.